### ISSUE TITLE
Multiple data_path and source image still can't be resolved

### ISSUE THREAD

**Issue by pzaj2** [liip/LiipImagineBundle#903]
| Q | A
| --- | ---
| Bug Report? | yes
| Feature Request? | no
| BC Break Report? | no
| RFC? | no
| Imagine Bundle Version | 1.7.2

I'm having a hard time using this bundle since 1.7 update. I've been able, following other issues' answers to resolve issue with symlinks, at least that's what I thought. I'm currently experiencing quite opposite issue.

I'm using Deployer PHP to deploy.

this is my loaders part of config:

```yml
liip_imagine:
    loaders:
        default:
            filesystem:
                data_root: "%liip_imagine_data_path%"
```

and my parameter is set that way:

```yml
liip_imagine_data_path:
        - '%kernel.root_dir%/../../../shared/web/'
        - '%kernel.root_dir%/../web/'
```
It seems these paths are resolved correctly, because the exception being logged is as follows:
```php
"Source image could not be found" at /var/www/releases/28/vendor/liip/imagine-bundle/Controller/ImagineController.php line 101 {
	"exception":"(...): 
		Source image could not be found at /var/www/releases/28/vendor//liip/imagine-bundle/Controller/ImagineController.php:101, 
			(...)NotLoadableException(code: 0): 
				Source image not resolvable \"images/application-octet-stream.png?v=1490184091182\" in root path(s) \"/var/www/shared/web:/var/www/releases/28/web\" at /var/www/releases/28/vendor/liip/imagine-bundle/Binary/Locator/FileSystemLocator.php:52)"} []
```

The problem is that there's a file
"/var/www/releases/28/web/images/application-octet-stream.png" and because of that I believe it should be resolvable, meaning it's either something I don't understand here or there's a bug somewhere.

**Comment by robfrawley** [comment]
The image you are requesting, per the exception message, is `images/application-octet-stream.png?v=1490184091182` whereas the file you want to load is `images/application-octet-stream.png`.

**Comment by pzaj2** [comment]
@robfrawley I understand and I did test it, you are right about that, I suppose my best shot would be to remove the twig all to asset function in order to strip the version parameter. Or perhaps there's some sort of a "hack", or parameter, to ignore certain parts of request URL?

**Comment by robfrawley** [comment]
Based on a cursory examination of the code, I do not see any special handling or configuration options that would allow it to ignore the query string section of a path. A pull request enabling such would be welcome.

**Comment by rpkamp** [comment]
Paths do not have query strings, URLs do. It seems you are mixing the two.
@pzaj2 could you show the twig code that caused this exception?

**Comment by rpkamp** [comment]
Issue seems related to https://github.com/liip/LiipImagineBundle/issues/838

**Comment by pzaj2** [comment]
@rpkamp Yes, this is the exact issue I am having. Well, I did simply remove asset helper calls followed by imagine filter, but I guess it would be good to have both co-existing well. For now I would close this bug report and focus on related issue

**Comment by robfrawley** [comment]
@rpkamp I understand paths cannot have query strings, but paths are passed via URLs, which can, hence they should likely be removed when the URL is converted into a path.

**Comment by rpkamp** [comment]
Or maybe the query string should be re-added to the URL? They were there for a reason, namely versioning.

So

`{{ 'path/to/image.png?v=123'|imagine_filter }}`

would result in something like

`/image/cache/resolve/image.img?v=123`

which will then in turn redirect to

`/image/cache/image.png?v=123`?

**Comment by dbu** [comment]
this has been partially fixed by stripping the query string. that brings its own can of worms, see #1392
